Removes lack of xenomorphs

TobyThorne opened the pull request:
Ксеноморфы в том виде, в каком они должны быть на бее.

- Добавлены лицехваты:
Появляются из яиц. Достать лицехвата вручную можно раньше, чем он вылупится сам.
Обнаружив человека, они пытаются подойти ближе, после чего прыгают, срывая шлемы и маски. Защитить могут герметичные шлемы. После промаха несколько секунд приходят в себя, готовясь к новому прыжку.
При большом желании и наличии открытого пространства от прыжка можно увернуться.
Если же лицехват присосался к лицу - жертва на какое-то время теряет сознание, а лицехват вживляет в жертву эмбрион ксеноморфа.
Лицехватов можно прилеплять к людям вручную, после небольшой задержки.

- Добавлены эмбрионы ксеноморфов:
Появляются в людях после изнасилования лицехватом. Удаляются из груди как обычные органы.
Если не удалить за четыре минуты - эмбрион разорвёт космонавтика на части и наружу вылезет грудолом.

- Грудоломы (larva) ксеноморфов:
У них почти ничего не изменилось. После появления грудолома призраки получают в чат уведомление, через которое могут занять грудолома.
Помимо сообщения в чате вселиться можно простым кликом ЛКМ.

- Ксеноморфы ксеноморфов:
- Теперь используют оригинальные спрайты вместо "выпрямленных".
- У ксеноморфов больше нет фантомных частей тел.
- Больше не чувствуют боль.
- Больше не могут поднимать предметы (за исключением мобов).
- Больше не могут использовать технику (возможно).
- Теперь ксеноморфы видят над космонавтиком значок, если в нём находится эмбрион.
- Конечности больше не ломаются. 
- Все ксеноморфы теперь "нейтрального" пола.
- Теперь видят в темноте.
- Очень сильно увеличено лечение на ксеноморфовской слизи или в атмосфере с плазмой. Защитники восстанавливаются в полтора раза быстрее, королавы - вдвое. Лежание (через Rest) ускоряет лечение втрое.
- Потерянные конечности и разрушенные внутренние органы теперь могут восстанавливаться, если находиться на слизи/в плазме. Срабатывает только если на остальных органах не осталось урона. Шанс - 1% за тик, если лежать - 10% за тик.
- Ксеноморфы теперь получают соответствующую роль. Уникальные задачи в данный момент ещё не назначены.
- Ксеноморфы теперь могут прорываться через шлюзы. Но медленно. И **очень** громко.
- Ксеноморфы теперь могут рушить столы.
- Ксеноморфы теперь могут бить ботов.
- Ксеноморфы больше не пачкают ноги в крови, ибо выглядит как говнище.
- Исправлена невозможность ползать по вентиляции.
- В кабинет РД на Исходе возвращена Ламарр.
- В кабинет РД на Фронтире добавлена Ламарр.

🆑
tweak: Ксеноморфы были полностью переработаны и добавлены в ротацию случайных ивентов. Больше информации на <a href="https://wiki.ss13.ru/index.php?title=Xenomorph">Wiki</a> и <a href="https://github.com/ChaoticOnyx/OnyxBay/pull/4949">GitHub</a>.
/🆑

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

TobyThorne committed:
xenorework

TobyThorne committed:
part 2

TobyThorne committed:
part 3

TobyThorne committed:
part 4

TobyThorne commented:
компилируйся сука

Zert0X commented on code/game/objects/effects/aliens.dm:
```diff
@@ -78,85 +76,55 @@
+	flick("egg_opening", src)
+	sleep(5)
+	if(get_turf(src))
+		new/mob/living/simple_animal/hostile/facehugger(get_turf(src))
```
А чё так можно было?

Doster-d replied:
Нельзя

TobyThorne committed:
part 5

TobyThorne committed:
part 5

TobyThorne committed:
part 7

TobyThorne committed:
part 8 (6)

TobyThorne committed:
part 9

TobyThorne commented:
Ламарр вернулась в кабинет РД!

TapKo4eB commented:
привет тестер тут
- фейсхаггеры проходят сквозь стены
- фейсхаггеры не прыгают на спящих
- фейсхаггеры повторно прыгают за заражённых
- лярвы спавнятся чистыми и становятся красными к 150/200 прогресса (должно быть наоборот?)
- лярвы медленные
- лярвы медленно заползают в вентиляцию
- лярвы не могут проползать под дверьми на лкм
- венты не видно под резиной
- алиены могут атаковать друг-друга
- алиены не могут атаковать клозеты
- резин мембраны и стены сносятся с одного удараа алиена
- резин двери не сносятся на харме
- трава не растёт под мусорками
- алиены не ускоряются на резине
- спрайт резин-дверей значительно отличается от стен и мембран
- плейсхолдеры на кнопках здоровья, блока и реста
- недоиконка в индикаторе огня у лярвы (нету у других алиенов), не хватает некоторых других (пикрелейтед)
- сломан регургигейт
- сломан верб коррозив эсид, работает не на клике
- сломан такл, работает не на клике
- сломан лип, работает не на клике
- сломан спит, не в той категории, работает не на клике
- тоггл асид спит хуй пойми как работает
- сломан трансфер плазма, в селекте есть хуманы и не работает проверка расстояния
- дарквижн не включен по дефолту
- асид ниже по слою чем дверь
- вскрытие болтированной двери руками не учитывается при отображении в чат
- нету *hiss
- *scream у алиенов на звук как крики трапа
- ломание двери сопровождается каким-то не тем звуком который должен быть
- вскрытие сломанной двери идёт без сообщений и звука
- вместо вскрытия алиены бьют обесточенную дверь
- жижа наноботов превращает алиенов(фи4а?)
- 
- алиены спавнятся с фулл плазмой(?)
- несты не ускоряют прогресс эмбриона(?)
- первый заражающий фейсхаггер не кнокает моба, каждый последующий успешно кнокает(?)
- фейсхаггеры детектят хуманов через стены(?)

![picrelated](https://user-images.githubusercontent.com/48761132/115544200-f8f6dc80-a2aa-11eb-92ce-09bf1f6b0719.png)

TobyThorne committed:
fix pack 1

- 'slash' sound is now used for slashing doors, lights and cameras;
- improves xenos-airlocks interactions;
- trying to interact w/ locked closets as a xenomorph no longer produces two messages;
- facehuggers no longer jump through solid objects (hopefully);
- facehuggers no longer leap on infected beings;
- xenomorphs no longer scream w/ girly voices;

TobyThorne commented:
Пачка фиксов номер раз:
- При ломании шлюзов, камер и лампочек теперь используется правильный звук ('slash");
- Улучшено взаимодействие ксеноморфов со шлюзами. Теперь выводятся сообщения о болтах/заваренных шлюзах, обесточенные шлюзы открываются без необходимости в предварительном избиении и всякое по мелочи;
- Ксеноморфы больше не полчают сразу два сообщения о lack dexterity при взаимодействии с закрытыми шкафчиками;
- Лицехваты больше не проходят через твёрдые объекты, **если те не пропускают воздух**. Если объект не герметичный - лицехват сможет на него запрыгнуть, но перед следующим прыжком будет кулдаун как при промахе;
- Лицехваты больше не прыгают на уже заражённых существ. Фича спорная, надо будет тщательнее обдумать;
- Ксеноморфы больше не кричат женскими голосами. Собственно, вообще никто с полом NEUTER не орёт женскими голосами. Только женщины;

TobyThorne committed:
fix pack part 2

- facehuggers can no longer leap through glass panes and windoors;
- facehuggers now apply paralyse properly;
- fixes lights smashing spam;

TobyThorne commented:
Ещё фиксов:
- Лицехваты больше не могут прыгать через тонкие окошки и виндуры;
- Лицехваты теперь правильно вырубают космонавтиков;
- Исправлен спам по уже разбитым лампочкам;

TobyThorne committed:
weeds (nonsmokable)

- Temporary changes weed and resin icons;
- Fixes weed spreading;

TobyThorne commented:
Траво-резиновое:
- Временно изменены спрайты конструкций и травы ксеноморфов. На классическе, но перекрашенные в более тёмный оттенок. В будущем, скорее всего, верну современные, но сейчас это решает кучу проблем. 
- Исправлено распространение травы, теперь она проникает повсюду, где могут проходить газы (следовательно, всякие закрытые шлюзы, окна и надувные стенки всё ещё блокируют рост);

Надо будет полностью переписать код травы, но уже в отдельном ПРе, наверное. А то этот и так блоутится как пиздец.
![image](https://user-images.githubusercontent.com/45202681/115610815-2f644400-a30b-11eb-9671-e766ea5ae20d.png)

TobyThorne committed:
hunger fix

TobyThorne committed:
Update alien_species.dm

TobyThorne committed:
icons

- Adds missing rest, block and blockswitch icons for aliens

TobyThorne committed:
facehugger window-warpers fix

TobyThorne commented:
- Окончательно пофикшена возможность лицехватов прыгать через окна по диагонали. Возможно. В теории, под определённым углом баг всё ещё может воспроизвестись, но вероятность крайне мала.
- Слегка улучшено взаимодействие ксенотравы с окнами. В некоторых случаях она всё ещё может прорасти, но с нынешней реализацией - едва ли я могу это хоть как-то поправить. Исправлю в будущем ПРе с переписыванием травы from scratch.

TobyThorne committed:
abilities stuff

TobyThorne commented:
800 строк? А может лучше 1100?
- Изменена работа способностей ксеноморфов. Теперь помимо вёрбов для непосредственного их использования есть вёрбы по типу "Set Tackle", устанавливающие выбранную способность как активную. Активная способность отображается во вкладке Status и может применяться выбранным в префе "Special Ability Activation Key" способом (по умолчанию - колёсико мыши). Поддерживаемые в данный момент способности - Tackle, Leap, Neurotoxic Spit, Acid Spit;
- Старые вёрбы для использования способностей через выбор из списка оставлены. Мало ли, может кому-то так удобнее;
- Аналогичный функционал теперь используют и воксы, для способности Leap;
- Добавлен вёрб "Disable Abilities", сбрасывающий активную способность. Может быть полено для тех, у кого модули РИГа и способности установлены на одну и ту же кнопку;
- Королевам ксеноморфов добавлены способности Tackle и Gut;
- Стоимость плевка увеличена с 20 до 25 плазмы;
- Урон от кислотного плевка увеличен с 30 до 45;
- Урон токсинами от нейротоксичного плевка увеличен с 5 до 10;
- Transfer Plasma теперь проверяет на наличие рабочего плазмаоргана у цели;

TobyThorne committed:
health changes

TobyThorne commented:
Измененения в здоровье и броне ксеноморфов:
- Раньше у всех типов были модификаторы к получаемому урону в 50% для брута и 200% для бёрна, стало:
- Дроны: 85% и 175%;
- Охотники: 75% и 165%;
- Стражи: 65% и 155%;
- Королева: 50% и 125%;
- Попутно увеличено количество здоровья у частей тел;

TobyThorne commented:
![image](https://user-images.githubusercontent.com/45202681/115747900-7f98e000-a3b7-11eb-9f48-0cca4d096145.png)
Думаю, на этом можно пока что остановиться и ревьювать. Состояние, в общем-то, достаточно играбельное.

Что осталось сделать:
- Переписать код травы. Нынешний - рабочий, но уж больно страшный. И оставляет гипотетическую возможность роста травы через окна под некоторыми углами. Но это уже в отдельном ПРе, со свежей головой и всякое такое.
- Что-то придумать с применением способностей через вёрбы. Сейчас они выдают полный список мобов/объектов поблизости. В лоб это не решается. Впрочем, это не особо мешает, ибо всё можно применять либо через контекстное меню, либо с помощью хоткея для способностей.
- Потестить всё с игроками и добалансировать всякие магические циферки.

TobyThorne committed:
final preparations

Doster-d commented on code/game/objects/items.dm:
```diff
@@ -592,9 +592,10 @@ var/list/global/slot_flags_enumeration = list(
 		if(!strong_knock)
 			H.visible_message(SPAN("warning", "[H]'s [src] flies off!"))
-			return
+			return 1
```
```suggestion
			return TRUE
```

Doster-d commented on code/game/objects/items.dm:
```diff
@@ -592,9 +592,10 @@ var/list/global/slot_flags_enumeration = list(
+			return 1
 	H.visible_message(SPAN("warning", "[H] falls down, unable to keep balance!"))
 	H.apply_effect(3, WEAKEN, 0)
+	return canremove ? 1 : 0
```
```suggestion
	return canremove ? TRUE : FALSE
```

Doster-d commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -671,7 +674,7 @@
 	if(species.has_fine_manipulation && !nabbing)
 		return 1
```
```suggestion
		return TRUE
```

Doster-d commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -671,7 +674,7 @@
 	if(!silent)
-		to_chat(src, "<span class='warning'>You don't have the dexterity to use that!</span>")
+		to_chat(src, FEEDBACK_YOU_LACK_DEXTERITY)
 	return 0
```
```suggestion
	return FALSE
```

Doster-d commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1,99 +1,176 @@
+		if(HUMAN_POWER_TACKLE)
+			var/mob/living/M = A
+			tackle(M)
+	return 1
```
```suggestion
	return TRUE
```

Doster-d commented on code/modules/mob/living/carbon/human/species/xenomorphs/alien_powers.dm:
```diff
@@ -46,34 +52,46 @@
 			return 0
 
 	P.stored_plasma -= cost
 	return 1
```
```suggestion
	return TRUE
```

Doster-d commented on code/modules/mob/living/carbon/human/species/xenomorphs/alien_species.dm:
```diff
@@ -44,135 +46,167 @@
 /datum/species/xenos/can_understand(mob/other)
-
 	if(istype(other,/mob/living/carbon/alien/larva))
 		return 1
```
```suggestion
		return TRUE
```

Doster-d commented on code/modules/mob/living/simple_animal/hostile/facehugger.dm:
```diff
@@ -0,0 +1,176 @@
+	H.equip_to_slot(holder, slot_wear_mask)
+	if(impregnate(H))
+		holder.kill_holder()
+	return 1
```
```suggestion
	return TRUE
```

Doster-d commented on code/modules/mob/living/simple_animal/hostile/facehugger.dm:
```diff
@@ -0,0 +1,176 @@
+		is_sterile = TRUE
+		to_chat(H, "You feel something going down your throat!")
+		H.Paralyse(20)
+		return 1
```
```suggestion
		return TRUE
```

Doster-d commented on code/modules/organs/external/xenos.dm:
```diff
@@ -73,27 +73,146 @@
-			GLOB.xenomorphs.add_antagonist_mind(H.mind,1, GLOB.xenomorphs.faction_role_text, GLOB.xenomorphs.faction_welcome)
+			GLOB.xenomorphs.add_antagonist_mind(H.mind, 1, GLOB.xenomorphs.faction_role_text, GLOB.xenomorphs.faction_welcome)
 
 	return 1
```
```suggestion
	return TRUE
```

Doster-d commented on code/modules/organs/external/xenos.dm:
```diff
@@ -73,27 +73,146 @@
+
+/obj/item/organ/internal/alien_embryo/Topic(href, href_list)
+	if(..())
+		return 1
```
```suggestion
		return TRUE
```

Doster-d commented on code/modules/projectiles/projectile/special.dm:
```diff
@@ -195,3 +195,64 @@
+		if(H.faction != holder.faction && holder.facefuck(H))
+			holder = null
+			qdel(src)
+			return 1
```
```suggestion
			return TRUE
```

Doster-d commented on code/modules/projectiles/projectile/special.dm:
```diff
@@ -195,3 +195,64 @@
+	set_density(0)
+	set_invisibility(101)
+	qdel(src)
+	return 1
```
```suggestion
	return TRUE
```

Doster-d commented on code/modules/mob/living/carbon/alien/alien.dm:
```diff
@@ -45,7 +45,6 @@
-	stat(null, "Progress: [amount_grown]/[max_grown]")
 
 /mob/living/carbon/alien/restrained()
 	return 0
```
```suggestion
	return FALSE
```

Doster-d commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -538,3 +538,6 @@
 		O.rejuvenate()
+
+/mob/living/carbon/proc/set_species()
+	return 0
```
```suggestion
	return FALSE
```

Doster-d commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1,99 +1,176 @@
-/mob/living/carbon/human/proc/tackle()
+/mob/living/carbon/human/proc/use_human_ability(atom/A)
+	if(!isliving(A))
+		return 0
```
```suggestion
		return FALSE
```

Doster-d commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1,99 +1,176 @@
+		return 0
+	switch(active_ability)
+		if(HUMAN_POWER_NONE)
+			return 0
```
```suggestion
			return FALSE
```

Doster-d commented on code/modules/mob/living/carbon/human/species/xenomorphs/alien_powers.dm:
```diff
@@ -12,30 +13,35 @@
-		to_chat(src, "<span class='danger'>You don't have enough plasma stored to do that.</span>")
+		if(!silent)
+			to_chat(src, SPAN("danger", "I don't have enough plasma stored to do that."))
 		return 0
```
```suggestion
		return FALSE
```

Doster-d commented on code/modules/mob/living/carbon/human/species/xenomorphs/alien_powers.dm:
```diff
@@ -46,34 +52,46 @@
-			to_chat(src, "<span class='danger'>You need a solid foundation to do that on.</span>")
+			if(!silent)
+				to_chat(src, SPAN("danger", "I need a solid foundation to do that on."))
 			return 0
```
```suggestion
			return FALSE
```

Doster-d commented on code/modules/mob/living/carbon/human/species/xenomorphs/alien_species.dm:
```diff
@@ -44,135 +46,167 @@
 	if(istype(other,/mob/living/carbon/alien/larva))
 		return 1
-
 	return 0
```
```suggestion
	return FALSE
```

Doster-d commented on code/modules/mob/living/carbon/human/species/xenomorphs/alien_species.dm:
```diff
@@ -44,135 +46,167 @@
+					if(W.wound_damage() == 0)
+						E.wounds -= W
+						return 1
 	return 0
```
```suggestion
	return FALSE
```

Doster-d commented on code/modules/mob/living/simple_animal/hostile/facehugger.dm:
```diff
@@ -0,0 +1,176 @@
+	var/obj/item/organ/external/chest/VC = H.organs_by_name["chest"]
+	var/obj/item/organ/external/head/VH = H.organs_by_name["head"]
+	if(!VC || !VH)
+		return 0
```
```suggestion
		return FALSE
```

Doster-d commented on code/modules/mob/living/simple_animal/hostile/facehugger.dm:
```diff
@@ -0,0 +1,176 @@
+	var/obj/item/helmet = H.get_equipped_item(slot_head)
+	if(helmet && ((helmet.item_flags & ITEM_FLAG_AIRTIGHT) || !helmet.knocked_out(H, dist = 0)))
+		H.visible_message(SPAN("notice", "\The [src] [pick("smacks", "smashes", "blops", "bonks")] against [H]'s [helmet] harmlessly!"))
+		return 0
```
```suggestion
		return FALSE
```

Doster-d commented on code/modules/mob/living/simple_animal/hostile/facehugger.dm:
```diff
@@ -0,0 +1,176 @@
+	var/obj/item/mask = H.get_equipped_item(slot_wear_mask)
+	if(mask && !mask.knocked_out(H, dist = 0))
+		H.visible_message(SPAN("warning", "\The [src] tries to rip [H]'s [mask] off, but fails."))
+		return 0
```
```suggestion
		return FALSE
```

Doster-d commented on code/modules/mob/living/simple_animal/hostile/facehugger.dm:
```diff
@@ -0,0 +1,176 @@
+
+/mob/living/simple_animal/hostile/facehugger/proc/impregnate(mob/living/carbon/human/H)
+	if(!H)
+		return 0
```
```suggestion
		return FALSE
```

Doster-d commented on code/modules/mob/living/simple_animal/hostile/facehugger.dm:
```diff
@@ -0,0 +1,176 @@
+	var/obj/item/organ/internal/alien_embryo/AE = H.internal_organs_by_name[BP_EMBRYO]
+	if(!AE)
+		if(is_sterile)
+			return 0
```
```suggestion
			return FALSE
```

Doster-d commented on code/modules/mob/living/simple_animal/hostile/facehugger.dm:
```diff
@@ -0,0 +1,176 @@
+		H.Paralyse(20)
+		return 1
+	to_chat(H, "You feel something going down your throat and rapidly ejecting a few moments later.")
+	return 0
```
```suggestion
	return FALSE
```

Doster-d commented on code/modules/mob/living/simple_animal/hostile/hostile.dm:
```diff
@@ -133,7 +134,7 @@
 	if(next_move >= world.time)
 		return 0
```
```suggestion
		return FALSE
```

Doster-d commented on code/modules/organs/external/xenos.dm:
```diff
@@ -73,27 +73,146 @@
 
-/obj/item/organ/internal/xenos/hivenode/replaced(mob/living/carbon/human/target,obj/item/organ/external/affected)
+/obj/item/organ/internal/xenos/hivenode/replaced(mob/living/carbon/human/target, obj/item/organ/external/affected)
 	if(!..()) return 0
```
```suggestion
	ASSERT(..())
```

TobyThorne replied:
тут ассерт нахуй не нужен, оно где-то там легально может нолик выдать

Doster-d replied:
Окей, тогда так.

TobyThorne replied:
![image](https://user-images.githubusercontent.com/45202681/115753395-fa182e80-a3bc-11eb-9e20-fb484f81ac87.png)

Doster-d reviewed (changes requested):
Ну, вот и ревью.

TobyThorne commented:
привет ты чё охуел

TobyThorne committed:
CYKA

Co-authored-by: Doster <63081538+Doster-d@users.noreply.github.com>

TobyThorne committed:
fuck

you

TobyThorne committed:
post-final tweaks

- Larvae growth time reduced from 200 seconds to 180;
- Spit cooldown increased from 1 second to 2 seconds;
- Xenomorphs can no longer spawn w/ neural laces installed;
- Xenomorphs can no longer have their faces disfigured;
- Neurotoxical spit TOX damage reduced from 10 to 5;

TobyThorne commented:
- Время для роста из грудолома во взрослого ксеноморфа уменьшено с 200 секунд до 180;
- Кулдаун плевков увеличен с 1 секунды до 2;
- Ксеноморфы больше не могут спавниться с установленным нейролейсом (касается только кривого админспавна);
- Лица ксеноморфов больше не могут быть изуродованы до неузнаваемости уроном. Unknown-ксеноморфы - это какой-то метамодерн;
- Урон токсинами у плевка нейротоксинами уменьшен с 10 до 5;

Doster-d reviewed (dismissed):
Ну я смотрел, локалку запускал, людей загонял, все норм.

BooJ1K commented:
Нашли пару приколов разной степени смешнявости:

- Фэйсхенгеры Всё ещё могут прыгать сквозь стекло, но так даже лучше
- Можно заражать взрослых Дион. По таймеру, они рассыпаются на мелких и высерают Лавру
- Можно заражать себя лицехватами и своих сородичей, взяв их в руки и ткнув по себе.
- Можно высаживать растения, будучи в вентиляции. 
- Невозможно регенить ожоги, в которых есть заражение. 
- Спутано цена (описание цены) за абилки Set acid spit Требует 50 тратит 25

![Кислоты](https://user-images.githubusercontent.com/47336771/115971649-dc40fa00-a573-11eb-8ffd-1e068b9eea19.png)

TobyThorne commented:
> * Спутано цена (описание цены) за абилки Set acid spit Требует 50 тратит 25
> 
> ![Кислоты](https://user-images.githubusercontent.com/47336771/115971649-dc40fa00-a573-11eb-8ffd-1e068b9eea19.png)

50 - трата на подготовку к плевкам, сами плевки тратят по 25, как и указано рядом со Spit.
Остальное сейчас пофикшу, да.

TobyThorne committed:
fix pack 3

TobyThorne commented:
- Исправлена скорость передвижения лицехватов;
- ФБП и дионы больше не могут быть заражены эмбрионом, но лицехваты всё равно будут пытаться;
- Ксеноморфы больше не могут быть заражены эмбрионом;
- Люди с вживлённым hivemind node ксеноморфа больше не атакуются лицехватами, а в случае случайного прыжка - не заражаются эмбрионом, но могут быть заражены им, если лицехвата посадить вручную;
- Грудоломы теперь видят в темноте;
- Способности вроде создания травы и кладки яиц теперь не могут быть применены из труб, шкафчиков, etc.
- Окончательно исправлено взаимодействие лицехватов и окон. Всё ещё есть один единственный случай, в котором лицехват может пройти через стекло, но с этим я уже ничего не могу поделать - это общая проблема взаимодействия окон и projectile'ов.

TobyThorne committed:
tiny thingy

TobyThorne committed:
specials

TobyThorne commented:
- Гнёзда теперь показывают оверлей поверх приклеенных жертв;
- ХП ксеноморфских дверей уменьшено до 200 (было 1500);
- Ксеноморфские двери теперь могут открывать только ксеноморфы и люди с вживлённым hivemind node;
- Ксеноморфы теперь ломают собственные постройки только на харме;
- Урон когтей и укусов ксеноморфов увеличен с 10 до 12.5;

- Добавлены особые ксеноморфы, получающиеся с помощью заражения определённых рас. Грудоломы, появившиеся из них всё ещё могут превратиться в любую касту, но имеют улучшенные варианты:
- Скрелл - может превратиться в Vile Drone вместо обычного. Имеет улучшенную регенерацию плазмы на траве и может плеваться кислотой;
- Унатх - может превратиться в Primal Sentinel вместо обычного. Имеет улучшенную регенерацию здоровья на траве и повышенную защиту от бёрна;
- Таяра - может превратиться в Feral Hunter вместо обычного. Имеет значительно повышенный урон от когтей (однако, урон от укусов остаётся прежним);
![alien_types](https://user-images.githubusercontent.com/45202681/116006765-79606900-a62e-11eb-9b62-e6daedcb3752.png)

Shootnik01 commented:
Крутяк какой!

TobyThorne committed:
Event

TobyThorne commented:
- Добавлен рандомный ивент "Stray Facehugger". Один лицехват спавнится в рандомной технической зоне.
- Добавлен рандомный ивент "Xenomorph Infestation". Спавнит грудоломов в определённых зонах. Количество зависит от онлайна, меньше 20 игроков - 1 грудолом, меньше 40 - два грудолома, иначе - три.

TobyThorne commented:
Теперь точно пора ревувать, да.

Shootnik01 commented:
Если лавру извлекают хирургически - она извлекается живой? Как хедкраб?

TobyThorne commented:
> Если лавру извлекают хирургически - она извлекается живой? Как хедкраб?

Извлекается не ларва, а эмбрион. При желании можно пересадить в кого-нибудь - мало ли, кто-нибудь вживит себе hivemind node и решит создать подконтрольную колонию ксеноморфов чтобы астероид зачищали.

Shootnik01 commented:
> > Если лавру извлекают хирургически - она извлекается живой? Как хедкраб?
> 
>мало ли, кто-нибудь вживит себе hivemind node и решит создать подконтрольную колонию ксеноморфов чтобы астероид зачищали.

Что за хайвмайнд нод?

TobyThorne committed:
microfix

Doster-d commented on code/modules/organs/external/xenos.dm:
```diff
@@ -73,27 +73,143 @@
 	owner.b_eyes = 153
 	..()
 
+process_sec_hud
```
Что это такое? Почему оно работает?

TobyThorne replied:
![image](https://user-images.githubusercontent.com/45202681/116246836-21d80f80-a78c-11eb-9eef-32ffbcda3054.png)
не знаю о чем ты

Doster-d commented:
cumфликты

TobyThorne committed:
Update xenos.dm

Doster-d reviewed (changes requested):
Все проверил, но cumфликт есть, после исправления будет зеленая галка.

TobyThorne committed:
Merge branch 'dev' of https://github.com/ChaoticOnyx/OnyxBay into analien

TobyThorne commented:
> Все проверил, но cumфликт есть, после исправления будет зеленая галка.

Cumфликты resolvлены.

TobyThorne committed:
gravity

TobyThorne committed:
evolve shortage

TobyThorne committed:
hekkofixes

TobyThorne committed:
Update lamarr.dm

Doster-d reviewed (dismissed):
Ну, вроде хорошо, даже локалку запускали под тесты.

Shootnik01 commented:
У тебя, походу. хедкрабы в вакууме дохнут

Shootnik01 commented:
Еще блоком от аллиенов на раз-два защититься, они его не пробивают

hard-res3t commented:
Нельзя снимать шлемы за алиенов.
Можно делать CPR людям, но говорят что фича.

TobyThorne committed:
Update facehugger.dm

TobyThorne commented:
Короче
- Лицехваты больше не умирают в вакууме;
- Лицехваты могут снимать герметичные шлемы, но только при надевании вручную;
- Лицехваты могут повторно заражать людей эмбрионом, если нынешний мёртв;

Shootnik01 commented:
Баг: Аллиены не ломают баррикады
Фича: кидать лицехватов в лицо врагам

Shootnik01 commented:
Еще момент: при росте из лавры в венте - ты появляешься под ней

Shootnik01 commented:
А еще они поинт ту не делают, что печально

TobyThorne committed:
barricades

TobyThorne committed:
Update progression.dm

TobyThorne commented:
- Убрал фон из-под индикатора здоровья ксеноморфов, а то заебали репортить;
- Ксеноморфы теперь могут ломать баррикады. Но барьеры СБ не могут, ибо нехуй;
- При эволюции внутри труб/шкафчиков/etc. моб не будет автоматически из них вылезать;

TobyThorne committed:
Update facehugger.dm

Shootnik01 commented:
Должны ли быть способности у хантера? Если да - то их не видно. если нет - зачем запас плазмы?
![image](https://user-images.githubusercontent.com/75479005/116451153-ce85c000-a864-11eb-94af-0c5aeaad82c2.png)

TobyThorne committed:
Update progression.dm

Shootnik01 commented:
Предложение: дать аллиенам возможность ломать обычные(не укрепленные) стены, ибо когти делают хрясь-хрясь

TobyThorne commented:
> Предложение: дать аллиенам возможность ломать обычные(не укрепленные) стены, ибо когти делают хрясь-хрясь

Нет

TobyThorne commented:
- Больше нельзя эволюционировать в трубах, ибо слишком проблемно. В шкафчиках-моргах-мешках - всё ещё можно.

Shootnik01 commented:
Кстати, лично мной не замеченное, но у пацанов вроде подтверждалось: человек с вживленным хранилищем плазмы не может его пополнять на траве(да и вообще). Может и фича, но тут хз.

TobyThorne committed:
objectives fix

TobyThorne commented:
- Исправлены обжективы у ксеноморфов. Теперь выполнение должно правильно засчитываться;

TobyThorne commented:
> Кстати, лично мной не замеченное, но у пацанов вроде подтверждалось: человек с вживленным хранилищем плазмы не может его пополнять на траве(да и вообще). Может и фича, но тут хз.

Фича. Хранить-то хранит, а всасывать нечем, нужно алиумотело.

TobyThorne committed:
Update xenomorph.dm

TobyThorne committed:
hiddenfix

Doster-d reviewed (approved):
Несколько раз гоняли на локалках, чуть в карты не поиграли, но ксеноморфы побили.
От себя хочу добавить, что если добавлять какой-то из режимов связанный с ксеноморфами, то стоит рассмотреть вариант: таторов + алиумов.

kreeperHLC reviewed (approved):
!!! ПР НЕ ПРОВЕРЯЛСЯ, ТОБИ ВЫПРОСИЛ ГАЛКУ ЧТОБЫ ФИКСИТЬ ВСЁ НА ХОДУ. ЕСЛИ ЧТО, ВИНИТЕ ТОБИ ВО ВСЕХ СМЕРТНЫХ ГРЕХАХ ПРА !!!

TobyThorne commented:
Короче, платный иссуй пока отлепляю отсюда. Закроем его вручную, когда убедимся, что всё работает как те самые часы.
А эта херня разрослась до гигантских размеров и грозит конфликтами. Остальные фичи и фиксы, если нужны будут, залью отдельно.

kreeperHLC commented:
Обкончал весь #github
![image](https://user-images.githubusercontent.com/14943437/116485665-71a1fe00-a894-11eb-9641-6a8129282d4f.png)
![image](https://user-images.githubusercontent.com/14943437/116485678-7666b200-a894-11eb-888c-5182566f105b.png)
